modules/seqexec/server/src/main/scala/seqexec/server/SequenceGen.scala

```diff
@@ -52,12 +53,12 @@ object SequenceGen {
     }
   }
 
-  final case class StepActionsGen[F[_]](pre: List[Actions[F]],
+  final case class StepActionsGen[F[_]](pre: List[ParallelActions[F]],
                                   configs: Map[Resource, Action[F]],
-                                  post: HeaderExtraData => List[Actions[F]]) {
-    def generate(ctx: HeaderExtraData): List[Actions[F]] =
+                                  post: HeaderExtraData => List[ParallelActions[F]]) {
+    def generate(ctx: HeaderExtraData): List[ParallelActions[F]] =
       pre ++
-        (if(configs.isEmpty) List() else List(configs.values.toList)) ++
+        NonEmptyList.fromList(configs.values.toList).foldMap(List(_)) ++
```

Comment:
Previously, you got a NEL from a `List` unsafely... how can you do it here without using the unsafely and make an assumption that `configs` is not an empty `Map`?

Reply:
I think we should avoid using the unsafe version and use something like this code instead. `foldMap` has a nice property that it will give you the `Monid` empty if needed. In this case `Nil`

Reply:
As suggested before this can be also `NonEmptyList.fromList(...).toList`